torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +280,48 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img.unbind(0)
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+
+    cr = maxc - minc
+    s = cr / maxc
+    rc = (maxc-r) / cr
+    gc = (maxc-g) / cr
+    bc = (maxc-b) / cr
+
+    t = (maxc != minc)
+    s = t * s
+    hr = (maxc == r) * (bc - gc)
+    hg = ((maxc == g) & (maxc != r)) * (2.0 + rc - bc)
+    hb = ((maxc != g) & (maxc != r)) * (4.0 + gc - rc)
+    h = (hr + hg + hb)
+    h = t * h
+    h = torch.fmod((h / 6.0 + 1.0), 1.0)
+    return torch.stack((h, s, maxc))
+
+
+def _hsv2rgb(img):
+    h, s, v = img.unbind(0)
+    i = torch.floor(h * 6.0)
+    f = (h * 6.0) - i
+    i = i.to(dtype=torch.int32)
+
+    p = torch.clamp((v * (1.0 - s)), 0.0, 1.0)
+    q = torch.clamp((v * (1.0 - s * f)), 0.0, 1.0)
+    t = torch.clamp((v * (1.0 - s * (1.0 - f))), 0.0, 1.0)
+    i = i % 6
+
+    mask = i[..., None] == torch.arange(6)
+    mask = mask.transpose(1, 2).transpose(0,1)
+
+    a1=torch.stack((v, q, p, p, t, v))
+    a2=torch.stack((t, v, v, q, p, p))
+    a3=torch.stack((p, p, t, v, v, q))
+    a4 = torch.stack((a1, a2, a3))
+
+    return torch.einsum("ijk, xijk -> xjk", mask.to(dtype=torch.float32), a4)
```

Comment:
instead of forcing the `mask` to be `float32` here, can you instead do (in like 319)
```
mask = mask.to(img)
```
so that it works for double tensors as well, as it will take the dtype of the `img` (and device)